frontend/src/contexts/dialogs.context.tsx

```diff
@@ -0,0 +1,149 @@
+} from "react";
+
+import { ConfirmDialog, ConfirmDialogProps } from "@/app-components/dialogs";
+import {
+  CloseDialog,
+  DialogComponent,
+  DialogProviderProps,
+  DialogStackEntry,
+  OpenDialog,
+  OpenDialogOptions,
+} from "@/types/common/dialogs.types";
+
+export const DialogsContext = createContext<
+  | {
+      open: OpenDialog;
+      close: CloseDialog;
+      confirm: FC<ConfirmDialogProps>;
+    }
+  | undefined
+>(undefined);
+
+/**
+ * Provider for Dialog stacks. The subtree of this component can use the `useDialogs` hook to
+ * access the dialogs API. The dialogs are rendered in the order they are requested.
+ *
+ * Demos:
+ *
+ * - [useDialogs](https://mui.com/toolpad/core/react-use-dialogs/)
+ *
+ * API:
+ *
+ * - [DialogsProvider API](https://mui.com/toolpad/core/api/dialogs-provider)
+ */
+function DialogsProvider(props: DialogProviderProps) {
+  const { children, unmountAfter = 1000 } = props;
+  const [stack, setStack] = useState<DialogStackEntry<any, any>[]>([]);
+  const keyPrefix = useId();
+  const nextId = useRef(0);
+  const requestDialog = useCallback<OpenDialog>(
+    function open<P, R>(
+      Component: DialogComponent<P, R>,
+      payload: P,
+      options: OpenDialogOptions<R> = {},
+    ) {
+      const { onClose = async () => {} } = options;
+      let resolve: ((result: R) => void) | undefined;
+      const promise = new Promise<R>((resolveImpl) => {
+        resolve = resolveImpl;
+      });
+
+      if (!resolve) {
+        throw new Error("resolve not set");
+      }
+
+      const key = `${keyPrefix}-${nextId.current}`;
+
+      nextId.current += 1;
+
+      const newEntry: DialogStackEntry<P, R> = {
+        key,
+        open: true,
+        promise,
+        Component,
+        payload,
+        onClose,
+        resolve,
+      };
+
+      setStack((prevStack) => [...prevStack, newEntry]);
+
+      return promise;
+    },
+    [keyPrefix],
+  );
+  const closeDialogUi = useCallback(
+    function closeDialogUi<R>(dialog: Promise<R>) {
+      setStack((prevStack) =>
+        prevStack.map((entry) =>
+          entry.promise === dialog ? { ...entry, open: false } : entry,
+        ),
+      );
+      setTimeout(() => {
+        // wait for closing animation
+        setStack((prevStack) =>
+          prevStack.filter((entry) => entry.promise !== dialog),
+        );
+      }, unmountAfter);
+    },
+    [unmountAfter],
+  );
+  const closeDialog = useCallback(
+    async function closeDialog<R>(dialog: Promise<R>, result: R) {
+      const entryToClose = stack.find((entry) => entry.promise === dialog);
+
+      if (!entryToClose) {
+        throw new Error("dialog not found");
+      }
+
+      await entryToClose.onClose(result);
+      entryToClose.resolve(result);
+      closeDialogUi(dialog);
+
+      return dialog;
+    },
+    [stack, closeDialogUi],
+  );
+  const contextValue = useMemo(
+    () => ({
+      open: requestDialog,
+      close: closeDialog,
+      confirm: ConfirmDialog,
```

Comment:
I'm not sure why are we passing a component in the context ?

Reply:
The mentioned code is fixed :white_check_mark: